DataFormats/simulation/include/SimulationDataFormat/MCTruthContainer.h

```diff
@@ -0,0 +1,98 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
+// distributed under the terms of the GNU General Public License v3 (GPL
+// Version 3), copied verbatim in the file "COPYING".
+//
+// See http://alice-o2.web.cern.ch/license for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+/// \file MCTruthContainer.h
+/// \brief Definition of a container to keep Monte Carlo truth external to simulation objects
+/// \author Sandro Wenzel - June 2017
+
+#ifndef ALICEO2_DATAFORMATS_MCTRUTH_H_
+#define ALICEO2_DATAFORMATS_MCTRUTH_H_
+
+#include <TNamed.h>
+#include <cassert>
+#include <stdexcept>
+
+namespace o2
+{
+namespace dataformats
+{
+// a simple struct having information about truth elements for particular indices:
+// how many associations we have and where they start in the storage
+struct MCTruthHeaderElement {
+  MCTruthHeaderElement() = default; // for ROOT IO
+
+  MCTruthHeaderElement(ushort s, uint i) : size(s), index(i) {}
+  ushort size = 0; // the number of entries
+  uint index = 0;  // the index into the actual MC track storage
+  ClassDefNV(MCTruthHeaderElement, 1);
+};
+
+// a container to hold and manage MC truth information
+// the actual MCtruth type is a generic template type and can be supplied by the user
+// It is meant to manage associations from one "dataobject" identified by an index into an array
+// to multiple TruthElements
+
+// note that we inherit from TObject just to be able to register this thing with the FairRootManager
+// (which might not be necessary in the future)
+template <typename TruthElement>
+class MCTruthContainer : public TNamed
+{
+ private:
+  std::vector<MCTruthHeaderElement>
+    mHeaderArray;                        // the header structure array serves as an index into the actual storage
```

Comment:
if you put the comment above the declaration, the member and type can be on one single line, I also suggest to start the comment with `///`, with the three slashes the comment makes it to the doxygen documentation. This is the same also for the comments before methods